File: src/my_project_typescript/easy_problems_typescript/from151to200/mostFrequentNumberFollowingKey.ts
function mostFrequent(nums: number[], key: number): number {

    let dic: {[key:number]:number} = []

    let ans: number = -1000000
    let temp: number = 0
    let lenNums: number = nums.length

    for (let i = 0; i < lenNums; i++){
        dic[nums[i]] = 0
    }

    for (let i = 0; i < lenNums-1; i++){
        if (nums[i] === key){
            dic[nums[i+1]] += 1
            if (temp < dic[nums[i+1]]){
                ans = nums[i+1]
                temp = dic[nums[i+1]]
            }
        }
    }

    return ans

};

File: src/my_project/easy_problems/from251to300/most_frequent_number_following_key.py
from typing import List 
import collections

class Solution:
    def mostFrequent(self, nums: List[int], key: int) -> int:
        dic = collections.defaultdict(int)
        ans = float("inf")
        temp = 0
        for i in range(len(nums)-1):
            if nums[i] == key:
                dic[nums[i+1]] += 1
                if temp < dic[nums[i+1]]:
                    ans = nums[i+1]
                    temp = dic[nums[i+1]]
        
        return ans
        
        
        